thunder/core/proxies.py

```diff
@@ -134,7 +134,7 @@ def replace_name(self, name: str | None = None):
         return self.__class__(name=name)
 
     def __repr__(self) -> str:
-        return f"{self.name}"
+        return f"<{type(self).__name__} {self.name}>"
```

Comment:
We can probably drop the <> brackets and include other attributes besides the name. For TensorProxies minimal valuable information is dtype and shape. Python's documentation suggests making `eval(repr(object))` work when possible, from [https://docs.python.org/3/reference/datamodel.html#object.__repr__](https://docs.python.org/3/reference/datamodel.html#object.__repr__):
> If at all possible, this should look like a valid Python expression that could be used to recreate an object with the same value (given an appropriate environment). If this is not possible, a string of the form <...some useful description...> should be returned.

Reply:
Thanks for the suggestion and information! Would something along the lines of `f"TensorProxy name=x, dtype=thunder.dtypes.float32, shape=(3,3)"` be sufficient? I think this will require a `prettyprint` update..

Reply:
I suggested the brackets, thanks Ivan for bringing this up, sounds great.

Reply:
We can also do this in a followup instead.

Reply:
Sure, it can happen in a followup. 
> Would something along the lines of f"TensorProxy name=x, dtype=thunder.dtypes.float32, shape=(3,3)" be sufficient?

The parentheses are missing, without parentheses it's not a valid Python expression: `TensorProxy(name="x", dtype=thunder.dtypes.float32, shape=(3,3))`.

Reply:
Thanks @IvanYashchuk @t-vi, let me update this by adding brackets, parentheses, dtype, and shape.